prow/config/branch_protection.go

```diff
@@ -278,20 +284,54 @@ func (c *Config) GetPolicy(org, repo, branch string, b Branch) (*Policy, error)
 		old, policy.Protect = policy.Protect, old
 		switch {
 		case policy.defined() && c.BranchProtection.AllowDisabledPolicies:
-			logrus.Warnf("%s/%s=%s defines a policy but has protect: false", org, repo, branch)
+			unprotectedRepo = true
 			policy = Policy{
 				Protect: policy.Protect,
 			}
 		case policy.defined():
-			return nil, fmt.Errorf("%s/%s=%s defines a policy, which requires protect: true", org, repo, branch)
+			return nil, false, fmt.Errorf("%s/%s=%s defines a policy, which requires protect: true", org, repo, branch)
 		}
 		policy.Protect = old
 	}
 
 	if !policy.defined() {
-		return nil, nil
+		return nil, false, nil
+	}
+	return &policy, unprotectedRepo, nil
+}
+
+// BranchProtectionWarnings returns two sets of strings:
+// - The list of repos with unprotected branches,
+// - The list of repos with disabled policies, i.e. Protect set to false,
+//     because any branches not explicitly specified in the configuration will be unprotected.
+func (c *Config) BranchProtectionWarnings() ([]string, []string) {
+	branchWarns := sets.NewString()
+	repoWarns := sets.NewString()
+	for orgName, org := range c.BranchProtection.Orgs {
+		for repoName, repo := range org.Repos {
+			repoRef := fmt.Sprintf("%s/%s", orgName, repoName)
+			repoPolicy := c.BranchProtection.GetOrg(orgName).GetRepo(repoName)
+			if repoPolicy.Protect != nil && !*repoPolicy.Protect {
+				if c.BranchProtection.AllowDisabledPolicies {
+					repoWarns.Insert(repoRef)
+				}
+			}
+			for branchName := range repo.Branches {
+				b, err := repoPolicy.GetBranch(branchName)
+				if err != nil {
+					continue
+				}
+				_, unprotectedRepo, err := c.getBranchPolicy(orgName, repoName, branchName, *b)
```

Comment:
Could we instead have a helper `isUnprotected(policy)` and post-process the policy we get?

Reply:
Thanks for the review. Done.